presto-main/src/main/java/com/facebook/presto/operator/project/InterpretedCursorProcessor.java

```diff
@@ -87,7 +88,7 @@ private static ExpressionInterpreter getExpressionInterpreter(
             parameterTypes.put(parameter, type);
         }
 
-        IdentityLinkedHashMap<Expression, Type> expressionTypes = getExpressionTypesFromInput(session, metadata, sqlParser, parameterTypes.build(), rewritten, emptyList());
+        IdentityLinkedHashMap<Expression, Type> expressionTypes = NodeRefCollections.toIdentityMap(getExpressionTypesFromInput(session, metadata, sqlParser, parameterTypes.build(), rewritten, emptyList()));
```

Comment:
why not also change `IdentityLinkedHashMap<Expression, Type> expressionTypes ` to
`Map<Node<? extends Expression>, Type> expressionTypes`?